aws/signer/v4/middleware.go

```diff
@@ -156,6 +156,16 @@ func (m *computePayloadSHA256) HandleBuild(
 	return next.HandleBuild(ctx, in)
 }
 
+// SwapComputePayloadSHA256ForUnsignedPayloadMiddleware replaces the
+// ComputePayloadSHA256 middleware with the UnsignedPayload middleware.
+//
+// Use this to disable computing the Payload SHA256 checksum and instead use
+// UNSIGNED-PAYLOAD for the SHA256 value.
+func SwapComputePayloadSHA256ForUnsignedPayloadMiddleware(stack *middleware.Stack) error {
+	_, err := stack.Build.Swap(computePayloadHashMiddlewareID, &unsignedPayload{})
+	return err
+}
```

Comment:
Don't think we should export this as standard util in the SDK. Unsigned payload doesn't work with all services, and is not a security best practice. Why is this required?

Reply:
This could be moved into S3 as a custom middleware generated there, but while its usage is specific to S3, its behavior could be used in other services that do support unsigned payload. Not sure S3.

Reply:
We can keep this for S3 and generate for services that use Unsignedpayload trait?

Reply:
Moving this to be generated by the relevant services is a option. 

This utility needs to be present in some form because the payload hash middleware ID nor any of the concrete middleware themselves.